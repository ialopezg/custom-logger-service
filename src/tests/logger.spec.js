const logger = require('../logger')

test('Must print a debugger message by a console', () => {
  const spy = jest.spyOn(logger, 'error').mockImplementation(() => {})

  const message = 'Console Error testing record'
  logger.error(message)

  expect(logger.error).toBeCalled()
  expect(spy.mock.calls[0][0]).toContain(message)
  expect(spy.mock.calls[0][0]).toMatchSnapshot()

  spy.mockRestore()
})

test('Must be an informative message', () => {
  const spy = jest.spyOn(logger, 'appStarted').mockImplementation(() => {})

  logger.appStarted(3000, 'localhost')

  expect(logger.appStarted).toBeCalled()
  expect(true).toMatchSnapshot()

  spy.mockRestore()
})
