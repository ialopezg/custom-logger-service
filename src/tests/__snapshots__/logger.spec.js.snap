// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must be an informative message 1`] = `true`;

exports[`Must print a debugger message by a console 1`] = `"Console Error testing record"`;
